src/main/java/org/folio/helper/PurchaseOrderLineHelper.java

```diff
@@ -808,11 +808,18 @@ private Future<Void> processPoLineEncumbrances(CompositePurchaseOrder compOrder,
         OrderWorkflowType.PENDING_TO_PENDING : OrderWorkflowType.PENDING_TO_OPEN;
       EncumbranceWorkflowStrategy strategy = encumbranceWorkflowStrategyFactory.getStrategy(workflowType);
       CompositePurchaseOrder poFromStorage = JsonObject.mapFrom(compOrder).mapTo(CompositePurchaseOrder.class);
-      return strategy.processEncumbrances(compOrder.withCompositePoLines(Collections.singletonList(compositePoLine)), poFromStorage, requestContext);
+      return strategy.processEncumbrances(compOrder.withCompositePoLines(Collections.singletonList(compositePoLine)), poFromStorage, requestContext)
+        .recover(throwable -> isInvoiceRelatedError(workflowType, throwable) ? Future.succeededFuture() : Future.failedFuture(throwable));
```

Comment:
We should not rely our business logic on handling error as in this code, if there is some invoice related error - we should not assume that edit was succesful, provided approach should be improved

Reply:
Done